src/test/util/matchable.rs

```diff
@@ -59,6 +59,20 @@ impl Matchable for Bson {
 impl Matchable for Document {
     fn content_matches(&self, expected: &Document) -> bool {
         for (k, v) in expected.iter() {
+            if k == "upsertedCount" {
```

Comment:
A more concise way would be to use the fact that `Option` is an iterator that always yields zero or one things; you could do `v == actual_v.iter().count()`

Reply:
I don't think that would work since there's a difference between `Some(Bson::Null)` and `None` here -- `None` indicates that the field doesn't exist, which means the documents don't match, whereas `Some(Bson::Null)` corresponds to `upsertedCount` being 0.

Reply:
Oh, good point; I didn't think of that